src/main/java/hobbylist/logic/parser/FindCommandParser.java

```diff
@@ -12,22 +13,43 @@
  * Parses input arguments and creates a new FindCommand object
  */
 public class FindCommandParser implements Parser<FindCommand> {
-
     /**
      * Parses the given {@code String} of arguments in the context of the FindCommand
      * and returns a FindCommand object for execution.
      * @throws ParseException if the user input does not conform the expected format
      */
     public FindCommand parse(String args) throws ParseException {
         String trimmedArgs = args.trim();
+        checkForFindRate(trimmedArgs);
         if (trimmedArgs.isEmpty()) {
             throw new ParseException(
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
         }
-
         String[] nameKeywords = trimmedArgs.split("\\s+");
-
         return new FindCommand(new NameOrDescContainsKeywordsPredicate(Arrays.asList(nameKeywords)));
     }
-
+    /**
+     * @throws ParseException if the input is empty, or containing strings, or rate value above 5 or below 0.
+     */
+    public void checkForFindRate(String trimmedArgs) throws ParseException {
+        if (trimmedArgs.contains("rate/")) {
+            String[] temp = trimmedArgs.split("/");
```

Comment:
I think this can cause a bug if you do something like `find cooking/eating rate/3`.

Reply:
I will fix this problem

Reply:
I am fixing this bug